tools/src/main/java/org/apache/kafka/tools/StorageTool.java

```diff
@@ -0,0 +1,478 @@
+                }
+            } else {
+                foundDirectories.add(directoryPath.toString());
+                Path metaPath = directoryPath.resolve("meta.properties");
+                if (!Files.exists(metaPath)) {
+                    problems.add(directoryPath + " is not formatted.");
+                } else {
+                    Properties properties = Utils.loadProps(metaPath.toString());
+                    RawMetaProperties rawMetaProperties = new RawMetaProperties(properties);
+                    Optional<RawMetaProperties> curMetadata;
+
+                    switch (rawMetaProperties.getVersion()) {
+                        case 0:
+                        case 1:
+                            curMetadata = Optional.of(rawMetaProperties);
+                            break;
+                        default:
+                            problems.add("Unsupported version for " + metaPath + ": " + rawMetaProperties.getVersion());
+                            curMetadata = Optional.empty();
+                            break;
+                    }
+
+                    if (!prevMetadata.isPresent()) {
+                        prevMetadata = curMetadata;
+                    } else {
+                        if (curMetadata.isPresent() && !prevMetadata.get().equals(curMetadata.get())) {
+                            problems.add(String.format("Metadata for %s was %s, but other directories featured %s", metaPath, curMetadata.get(), prevMetadata.get()));
+                        }
+                    }
+                }
+            }
+        }
+
+        if (prevMetadata.isPresent()) {
+            if (selfManagedMode) {
+                if (prevMetadata.get().getVersion() == 0) {
+                    problems.add("The kafka configuration file appears to be for a cluster in KRaft mode, but " + "the directories are formatted for legacy mode.");
+                }
+            } else if (prevMetadata.get().getVersion() == 1) {
+                problems.add("The kafka configuration file appears to be for a legacy cluster, but " + "the directories are formatted for a cluster in KRaft mode.");
+            }
+        }
+
+        return validateDirectories(stream, directories, problems, foundDirectories, prevMetadata);
+    }
+
+    private static int validateDirectories(PrintStream stream, List<String> directories, List<String> problems, List<String> foundDirectories, Optional<RawMetaProperties> prevMetadata) {
+        if (directories.isEmpty()) {
+            stream.println("No directories specified.");
+            return 0;
+        } else {
+            if (!foundDirectories.isEmpty()) {
+                if (foundDirectories.size() == 1) {
+                    stream.println("Found log directory:");
+                } else {
+                    stream.println("Found log directories:");
+                }
+                foundDirectories.forEach(d -> stream.printf(d + "%n"));
```

Comment:
In the old implementation we have 2 spaces before the log dir path. It is important to keep the same output, because there may be some tool parsing it that we don't want to break.